### DISCUSSION TITLE
Home assistant HyperHDR addon

### DISCUSSION THREAD

**Discussion by BozoLord** [awawa-dev/HyperHDR#525] [Q&A]
Hi
I found a HyperHDR addon on my Home assistant
when I installed it, it created a HyperHDR instance
I actually have HyperHDR running on a pi and Home assistant on a Dell optiplex
What would be the wiring to run HyperHDR from the HA instance on the optiplex ?

**Comment by awawa-dev** [comment]
Hi
So you want to run HyperHDR directly on Dell under HA instead of Rpi? Did you try https://github.com/ihrapsa/hassio-addons to install it? From Dell you can use HyperSerialEsp8266/HyperSerialESP32 (USB) or WLED if you prefer wifi connection to control the LEDs.

**Reply by awawa-dev** [reply]
>  HyperSerialEsp8266/HyperSerialESP32 (USB)

Just remember that since HyperHDR runs in docker under HA, the USB port probably needs to be redirected to docker. I don't use HA, but it's a rather standard topic with similar uses for other applications.

**Reply by BozoLord** [reply]
So if I use hyperSerialEsp8266, I plug the esp8266 and the video capture card into the PC
And the led strip to the esp8266, right ?

**Reply by awawa-dev** [reply]
Yes, this is a plan for a standard Linux/Windows PC. If you haven't purchased yet I recommend esp32 (list on HyperSerialESP32, ESP32-S2 won't work with WLED) instead of esp8266 and it's not much more expensive than go for it. But I don't use HyperHDR in a docker container inside HA, so I don't know if you can experience some performance drop, especially with USB grabber. Maybe other users here will comment if they have such a solution.

**Comment by BozoLord** [comment]
Got hyperSerialEsp32 flashed
Plugged it in my host
My video grabber is detected (I see the image of my TV in hyperhdr)
I plugged data cable into D2 pin
But my leds are always off
Is there a way to test the hyperSerialEsp32 ?

**Reply by awawa-dev** [reply]
D2? Are you sure? It's usually something different than GPIO2. Which board do you use? Have a pinout diagram for this model?

**Reply by BozoLord** [reply]
https://lastminuteengineers.b-cdn.net/wp-content/uploads/iot/ESP32-Pinout.png

GPIO2 is D2 on this board, right ?

**Reply by awawa-dev** [reply]
Yes, in this case it is. But this board isn't on the recommended list and have a special section in my latest post on the blog.
![obraz](https://user-images.githubusercontent.com/69086569/222794403-f2e47e00-592f-4cab-94bb-d70d3b06e46b.png)
Anyway first you need to check if there is a communication (at 2 000 000 speed if you used default firmware).
Follow this guide to test max. framerate.
https://www.hyperhdr.eu/2023/02/ultimate-guide-on-how-to-build-led.html#chapter9
Provide a log from HyperHDR from testing.

**Reply by d22b** [reply]
> Got hyperSerialEsp32 flashed
> Plugged it in my host
> My video grabber is detected (I see the image of my TV in hyperhdr)
> I plugged data cable into D2 pin
> But my leds are always off
> Is there a way to test the hyperSerialEsp32 ?

Highly recommend to use the „official“ supported esp32 boards. By using a differently pinout, you have to change the data pin in the hyperserial firmware by yourself. 
Tutorial: https://github.com/awawa-dev/HyperSerialESP32/wiki

**Comment by oywino** [comment]
Please excuse me for butting in - but is the HA HyperHDR addon a replacement for the official HyperHDR v19 - or is it an extension (integration) to connect HA to an already running instance of HyperHDR somewhere else in the network?
When I installed HyperHDR, HA immediately found it and after successful integration gave me a number of entities to play with.   Will this addon replace or extend these features, or am I completely lost?
